### PR TITLE
[4.x] Add support for brotli

### PR DESCRIPTION
**Pull request by Razzo1987**
Add support for brotli gz compression in htaccess.txt

From [Wikipedia](https://en.wikipedia.org/wiki/Brotli):
> Brotli is a [lossless data compression algorithm](https://en.wikipedia.org/wiki/Lossless_compression) developed by [Google](https://en.wikipedia.org/wiki/Google). It uses a combination of the general-purpose [LZ77](https://en.wikipedia.org/wiki/LZ77_and_LZ78) [lossless compression](https://en.wikipedia.org/wiki/Lossless_compression) algorithm, [Huffman coding](https://en.wikipedia.org/wiki/Huffman_coding) and 2nd-order [context modelling](https://en.wikipedia.org/wiki/Context_model). Brotli is primarily used by [web servers](https://en.wikipedia.org/wiki/Web_servers) and [content delivery networks](https://en.wikipedia.org/wiki/Content_delivery_networks) to [compress HTTP content](https://en.wikipedia.org/wiki/HTTP_compression), making internet websites load faster. A successor to [gzip](https://en.wikipedia.org/wiki/Gzip), it is supported by all major web browsers and has become increasingly popular, as it provides better compression than gzip.

### Summary of Changes
Webservers that use brotli and have Joomla's htaccess enabled will fail the website in HTTPS unless you comment/delete the code block related to GZIP.
After the change, the problem no longer occurs.

### Testing Instructions
1. Check if your hosting uses brotli.
2. Apply the patch and copy the text from `htaccess.txt` to `.htaccess`
3. Indicate in the test whether you tested with a hosting that uses brotli or not. I think it is useful to have tests for both cases.

#### If your hosting does not use brotli
Joomla works properly

#### If your hosting use brotli
Joomla in HTTP works correctly.
Joomla in HTTP**S** does not load the CSS and JS resources, and investigating the error in the console is: `ERR_CONTENT_DECODING_FAILED`

#### If your hosting does not use brotli
Joomla still works properly

#### If your hosting use brotli
Joomla works properly

### NOTE
[ServerPlan](https://www.serverplan.com/) uses brotli.

### Thanks
Thanks to ServerPlan for supporting the tests and finding the solution

### Link to documentations
Please select:
- [ ] Documentation link for docs.joomla.org: <link>
- [ ] No documentation changes for docs.joomla.org needed

- [ ] Pull Request link for manual.joomla.org: <link>
- [ ] No documentation changes for manual.joomla.org needed

### CODE REVIEW

File: htaccess.txt [hunk 1]
```diff
@@ -158,8 +158,8 @@ Options -Indexes
 	RewriteRule "^(.*)\.js" "$1\.js\.gz" [QSA]
 
 	# Serve correct content types, and prevent mod_deflate double gzip.
```
Comment (brianteeman) [inline htaccess.txt]: probably should update this comment as well

Comment (Razzo1987) [reply htaccess.txt]: I'm thinking about how to write it, because both gzip and brotli use DEFLATE to produce a .gz file ...
Suggestions are welcome :D

Comment (brianteeman) [reply htaccess.txt]:
```suggestion
	# Serve correct content types, and prevent mod_deflate double compression.
```

?

Comment (Razzo1987) [reply htaccess.txt]: good idea!

**Comment by MacJoom** [comment]
I have tested this item :white_check_mark: successfully on 33d2ff19f8ff5efa4f66e033c226542a1a902f01<br /><br />I have setup brotli compression on my own server (you need to disable gzip - and you need https!) and the change in the .htaccess fixed the issue with not loaded css and js. brotli see here: https://www.brotli.pro/enable-brotli/servers/apache/<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/43055">issues.joomla.org/tracker/joomla-cms/43055</a>.</sub>

**Comment by Razzo1987** [comment]
INFO: It seems that with ngix instead of apache the problem does not occur

**Comment by dgrammatiko** [comment]
This PR seems wrong according to the docs of [Apache](https://httpd.apache.org/docs/2.4/mod/mod_brotli.html) 

As stated the **brotli** compressed files are expected to have a `.br` extension and this PR just (?) assigns `.gz` to Brotli (?)
```
    # Serve brotli compressed JS files if they exist
    # and the client accepts brotli.
    RewriteCond "%{HTTP:Accept-encoding}" "br"
    RewriteCond "%{REQUEST_FILENAME}\.br" "-s"
    RewriteRule "^(.*)\.js"               "$1\.js\.br" [QSA]
```

In sort if the ISP expects that the brotli files are having `.br` extension then they are misconfigured...

My 2c anyways

**Comment by dgrammatiko** [comment]
I have tested this item :white_check_mark: successfully on 33d2ff19f8ff5efa4f66e033c226542a1a902f01<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/43055">issues.joomla.org/tracker/joomla-cms/43055</a>.</sub>

**Comment by alikon** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/43055">issues.joomla.org/tracker/joomla-cms/43055</a>.</sub>

**Comment by laoneo** [comment]
Thanks!

**Comment by MacJoom** [comment]
Hi  @brianteeman - we need a Post-Installation Message for the change in the htaccess.txt - as a native speaker could you suggest a message? As we are building the RC release this evening this is urgent. Sorry for the hurry. Please ping me if you are busy.

**Comment by brianteeman** [comment]
@MacJoom untested but should be ok - #43171

**Pull request merged**